### ISSUE TITLE
BatchSpanProcessor supports DefaultMaxExportPacketSize option

### ISSUE THREAD

**Issue by tianyaqu** [open-telemetry/opentelemetry-go#2415]
### Problem Statement
Users may append events and attributes to a span, that will make batched spans too large to export to  collector, complaing about
`ResourceExhausted=grpc: received message larger than max... `

The codes blow shows how users may add custom events and attributes to a span,
```
	rsp, err := doRPC(ctx, req)
	_, span := tracer.Start(ctx, "service-name")
	defer span.End()

	// add req & rsp body to span, large req&rsp bodies are not rare in real world.
	span.AddEvent("sent",
		trace.WithAttributes(attribute.Key("msg").String(req.String())))
	span.AddEvent("receive"),
		trace.WithAttributes(attribute.Key("msg").String(rsp.String())))
	
        // add other attributes
	span.SetAttributes(attribute.Key("net.host").String("1.2.3.4"))
	span.SetAttributes(attribute.Key("net.peer").String("4.5.6.7"))
        span.SetAttributes(attribute.Key("time.cost").String("121ms"))
	.....
	span.SetAttributes(attribute.Key("Region").String("us"))
```

### Proposed Solution
Adds a DefaultMaxExportPacketSize option to limit the size of batched spans.

Related to #2414

**Comment by Aneurysm9** [comment]
I agree with @MrAlias's comment on #2414 that the BSP is probably not the correct place to implement this.  The Jaeger exporter already has [a similar capability](https://github.com/open-telemetry/opentelemetry-go/blob/main/exporters/jaeger/agent.go#L126-L149) in its agent uploader.  I think that is really the only place that we can know whether the data to be emitted are larger than the transmission mechanism can handle.  If we attempt to do this in the BSP we won't be able to account for encoding overhead or compression savings.

It might be good to have a consistent option that can be set on exporters that can implement this sort of batch splitting directly, which would provide at least a consistent UX.

**Comment by MrAlias** [comment]
Thanks for opening this issue. As stated above, we do not plan to include this enhancement proposal in the project. Please feel free to address this at the exporter level if you think that would be appropriate.
